### ISSUE TITLE
Possible improvements to SHACL validation

### ISSUE THREAD

**Issue by AleksanderZawisza** [OpenCS-ontology/OpenCS#20]
- Currently the validation process is done during the pre-release push check (in file: https://github.com/OpenCS-ontology/OpenCS/blob/main/.github/workflows/pre-release.yml). It can be easily changed to be run on every pull request by modifying or creating new .yml file in workflows.
- The "Validate ontology" job utilizes the processing of the ontology used also in the "Package ontology" job. Subsequently the process of loading the ontology into RDFlib graph (which takes around 1min) is run twice (for both of these jobs). The time of execution of the whole workflow can be therefore reduced by loading the ontology once before both of these tasks.

**Comment by Ostrzyciel** [comment]
Could you split this into two separate issues? It will be much easier to track progress on both topics this way.

**Comment by Ostrzyciel** [comment]
Ok, nevermind, I resolved the first part, and the second is invalid now due to how I solved it (it was moved to a completely separate job to run in parallel).
